Formula/b/bandicoot.rb

```diff
@@ -0,0 +1,42 @@
+class Bandicoot < Formula
+  desc "C++ library for GPU accelerated linear algebra"
+  homepage "https://coot.sourceforge.io/"
+  url "https://gitlab.com/conradsnicta/bandicoot-code/-/archive/1.11.0/bandicoot-code-1.11.0.tar.bz2"
+  sha256 "e53fa0db2cd890475ad41452b0a70b036315e32dcbe45687f149789e341a85f6"
+  license "Apache-2.0"
+
+  livecheck do
+    url :stable
+    regex(%r{url=.*?/bandicoot-code[._-]v?(\d+(?:\.\d+)+)\.t}i)
+  end
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "clblas"
+  depends_on "openblas"
+
+  def install
+    # Enable the detection of OpenBLAS
+    system "cmake", ".", "-DALLOW_OPENBLAS_MACOS=ON", "-DALLOW_BLAS_LAPACK_MACOS=ON", *std_cmake_args
+    system "make", "install"
+  end
+
+  test do
+    # Create a test script that compiles a program
+    (testpath/"test.cpp").write <<~EOS
+      #include <iostream>
+      #include <bandicoot>
+
+      int main(int argc, char** argv) {
+        std::cout << coot::coot_version::as_string() << std::endl;
+      }
+    EOS
+    system ENV.cxx, "-std=c++11", "test.cpp", "-I#{include}", "-L#{lib}", "-lbandicoot", "-o", "test"
+
+    # Avoid running the test if in CI due to no GPU being available.
+    return if Hardware::CPU.arm? && MacOS.version >= :monterey && ENV["HOMEBREW_GITHUB_ACTIONS"].present?
+
+    # Check that the coot version matches with the formula version
+    assert_equal shell_output("./test").to_i, version.to_s.to_i
+  end
```

Comment:
We need some kind of test that can succeed in CI so that we can ensure the application runs successfully.

Reply:
@p-linnane so there are a couple of Formula's that are GPU-specific that omit testing on the CI system due to a lack of a GPU.

`clblas`: 
https://github.com/Homebrew/homebrew-core/blob/8a9fcbcf5cdb35aaa8e0a7e87892fa32a64bb77a/Formula/c/clblas.rb#L42

`arrayfire`: 
https://github.com/Homebrew/homebrew-core/blob/8a9fcbcf5cdb35aaa8e0a7e87892fa32a64bb77a/Formula/a/arrayfire.rb#L68-L69

If a GPU toolkit isn't detected, the library will not work. The goal for the library is to be a drop-in replacement for `armadillo` (CPU-bound) to allow for GPU calculations.

Reply:
That's fine if there's a precedent for it. Thanks for the heads up.

Reply:
I agree that it's quite strange that library cannot provide its version without GPU being on the system.
Our formula (openvino.rb) can also utilize GPU, but it checks for GPU presence and works perfectly if it's absent

Reply:
@ilya-lavrenov and @p-linnane I've removed the one `return` line and the CI is fine. (Outside of the notability check.) Hopefully, this can speed along the inclusion of the formula.